parsec/core/fs/workspacefs/workspacefs.py

```diff
@@ -198,18 +195,20 @@ def encryption_revision(self) -> int:
 
     # Timestamped version
 
+    async def list_versions(self, path: AnyPath = "/"):
+        path = FsPath(path)
+        try:
+            versions_dict = await self.entry_transactions.entry_versions(path)
+        except (FileNotFoundError, NotADirectoryError):
+            return {}
+        return versions_dict
+
     async def to_timestamped(self, timestamp: Pendulum):
         workspace = workspacefs.WorkspaceFSTimestamped(self, timestamp)
         try:
             await workspace.path_info("/")
-        except FSRemoteManifestNotFoundBadVersion:  # as exc:
-            # Temp hack...
-            fake_manifest = LocalWorkspaceManifest.make_placeholder(
-                author=self.device.device_id, parent_id=self.workspace_id, created=timestamp
-            )
-            workspace.local_storage.local_manifest_cache[self.workspace_id] = fake_manifest
-            # TODO : use this exception instead of mocking an empty fs when the API is ready
-            # raise FSWorkspaceTimestampedTooEarly from exc
+        except FSRemoteManifestNotFoundBadVersion as exc:
+            raise FSWorkspaceTimestampedTooEarly from exc
```

Comment:
Raise une instance de `FSWorkspaceTimestampedTooEarly` plutôt que la classe ?

Reply:
Il y a une différence si j'ai pas d'arguments à passer? sachant que je fais un raise from, donc j'ai normalement déjà accès aux infos dont j'ai besoin dans mon exception

Reply:
Je ne savais pas la différence entre raise une classe ou une instance du coup c'était pour savoir si t'en savais plus ;-)

Je viens de regarder la doc https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement:
> If it is a class, the exception instance will be obtained when needed by instantiating the class with no arguments.

Donc ça revient au même de faire `raise FSWorkspaceTimestampedTooEarly` ou `raise FSWorkspaceTimestampedTooEarly()` ^^